cylc/flow/scripts/lint.py

```diff
@@ -0,0 +1,546 @@
+        re.compile(r'\s$'): {
+            'short': 'trailing whitespace.',
+            'url': STYLE_GUIDE + 'trailing-whitespace'
+        },
+        # Consider re-adding this as an option later:
+        # re.compile(r'^.{80,}'): {
+        #     'short': 'line > 79 characters.',
+        #     'url': STYLE_GUIDE + 'line-length-and-continuation'
+        # },
+        re.compile(r'inherit\s*=\s*[a-z].*$'): {
+            'short': 'Family name contains lowercase characters.',
+            'url': STYLE_GUIDE + 'task-naming-conventions'
+        },
+    }
+}
+
+
+def parse_checks(check_arg):
+    """Collapse metadata in checks dicts.
+    """
+    parsedchecks = {}
+    if check_arg == '728':
+        purpose_filters = ['U']
+    elif check_arg == 'lint':
+        purpose_filters = ['S']
+    else:
+        purpose_filters = ['U', 'S']
+
+    for purpose, checks in CHECKS.items():
+        if purpose in purpose_filters:
+            for index, (pattern, meta) in enumerate(checks.items()):
+                meta.update({'purpose': purpose})
+                meta.update({'index': index})
+                parsedchecks.update({pattern: meta})
+    return parsedchecks
+
+
+def check_cylc_file(file_, checks, modify=False):
+    """Check A Cylc File for Cylc 7 Config"""
+    # Set mode as read-write or read only.
+    outlines = []
+
+    # Open file, and read it's line to mempory.
+    lines = file_.read_text().split('\n')
+    jinja_shebang = lines[0].strip().lower() == JINJA2_SHEBANG
+    count = 0
+    for line_no, line in enumerate(lines):
+        for check, message in checks.items():
+            # Tests with for presence of Jinja2 if no shebang line is
+            # present.
+            if (
+                jinja_shebang
+                and message['short'].startswith(
+                    JINJA2_FOUND_WITHOUT_SHEBANG)
+            ):
+                continue
+            if check.findall(line) and not line.strip().startswith('#'):
+                count += 1
+                if modify:
+                    if message['url'].startswith('http'):
+                        url = message['url']
+                    else:
+                        url = URL_STUB + message['url']
+                    outlines.append(
+                        f'# [{message["purpose"]}{message["index"]:03d}]: '
+                        f'{message["short"]}\n'
+                        f'# - see {url}'
+                    )
+                else:
+                    print(
+                        Fore.YELLOW +
+                        f'[{message["purpose"]}{message["index"]:03d}]'
+                        f'{file_}: {line_no}: {message["short"]}'
+                    )
+        if modify:
+            outlines.append(line)
+    if modify:
+        file_.write_text('\n'.join(outlines))
+    return count
+
+
+def get_cylc_files(base: Path) -> Generator[Path, None, None]:
+    """Given a directory yield paths to check.
+    """
+    excludes = [Path('log')]
```

Comment:
Should exclude processed files too

Reply:
Processed files are inside the log dir, so they don't get  scanned.

Reply:
I can no longer reproduce

Reply:
I've pushed a change - looks like `Path.parents` doesn't exactly do what I thought that it did.